service/stub_module.go

```diff
@@ -29,17 +29,17 @@ type StubModuleProvider struct {
 var _ ModuleProvider = &StubModuleProvider{}
 
 // NewDefaultStubModuleProvider returns a new StubModuleProvider with an empty StubModule.
-func NewDefaultStubModuleProvider(name string) *StubModuleProvider {
-	return NewStubModuleProvider(name, &StubModule{})
+func NewDefaultStubModuleProvider() *StubModuleProvider {
+	return NewStubModuleProvider("stubModule", &StubModule{})
 }
 
 // NewStubModuleProvider returns a new StubModuleProvider with the given name
 // and create function from NewStubModuleCreateFunc.
 func NewStubModuleProvider(name string, stubModule *StubModule) *StubModuleProvider {
-	return &StubModuleProvider{name, NewStubModuleCreateFunc(stubModule)}
+	return &StubModuleProvider{NameVal: name, CreateVal: NewStubModuleCreateFunc(stubModule)}
 }
 
-// DefaultName returns the default name.
+// DefaultName returns the module name
```

Comment:
The comment is incorrect, i've explained the function naming already. It returns the default, fallback, whatever you want to call it, if a name option is not explicitly provided.

I can't stress it enough how much i'm tired of talking about this, but can we please stop renaming things backwards and forwards and just move on? If proven confusing, easy to fix in the future. Lets stop guessing